### ISSUE TITLE
Unknown types in celt header file on macOS 12 ARM

### ISSUE THREAD

**Issue by thegcat** [mumble-voip/mumble#5474]
### Description

Building Mumble on macOS 12 ARM currently fails

### Steps to reproduce

1. Checkout `1.4.x` branch
2. `cmake -Dice=off ..`
3. `cmake --build . -j 4`

### Mumble version

09ad3a8d7

### Mumble component

Client

### OS

macOS

### Reproducible?

Yes

### Additional information

Notwithstanding the fact that celt seems to be folded into opus now, the standalone celt library has been updated since it has been included in mumble. It seems celt-0.7.1 would include the [following fix](https://gitlab.xiph.org/xiph/celt/-/commit/80cae17845605237cc2d408d17a07733b7e3635c), which in turn allow to build mumble on macOS ARM.

### Relevant log output

```shell
[ 64%] Building CXX object src/mumble/CMakeFiles/mumble.dir/AudioOutputSpeech.cpp.o
In file included from /Users/thegcat/Code/m1/mumble/src/mumble/CELTCodec.cpp:6:
In file included from /Users/thegcat/Code/m1/mumble/src/mumble/CELTCodec.h:9:
In file included from /Users/thegcat/Code/m1/mumble/3rdparty/celt-0.7.0-src/libcelt/celt.h:42:
/Users/thegcat/Code/m1/mumble/3rdparty/celt-0.7.0-src/libcelt/celt_types.h:70:12: error: unknown type name 'int16'
   typedef int16 celt_int16;
           ^
/Users/thegcat/Code/m1/mumble/3rdparty/celt-0.7.0-src/libcelt/celt_types.h:71:12: error: unknown type name 'u_int16'
   typedef u_int16 celt_uint16;
           ^
```

**Comment by Krzmbrzl** [comment]
Oh that's interesting - didn't know there has been an update to CELT. We used to fix this issue by defining the `HAVE_STDINT_H` macro, which causes the CELT header to use the definitions from the stdint.h header instead but that was causing other issues. It might provide a viable workaround though (until we get to update our CELT submodule)

**Comment by thegcat** [comment]
Patching it locally worked, and maybe just going with the latest dot release of celt-0.7.x would already help.

I'm not quite sure though why this happens on ARM and not Intel macOS? Does Intel macOS builds use the block just above with `defined(__MACOS__)` and for some reason that does not trigger on ARM macOS and it falls back to `defined(__APPLE__) && defined(__MACH__)`?

(My last contact with C/C++ was at university and I just applied `*` and `&` liberally until stuff worked, so I'm talking out of my ass here. Still wondering what the difference between ARM macOS and Intel macOS would be that the it would work on one and not the other system for this code)

**Comment by Krzmbrzl** [comment]
I don't know why it sometimes works and sometimes doesn't but it can also fail on x86  machines. An example of that is our macOS builder ^^

**Comment by amberhandal** [comment]
Hi, I had this issue when building on my Mac, I think I solved it by changing the type defs. I'm not sure if this solution is going to apply to older OSs though. I changed lines 70-71 in celt_types.h respectively to: 
` typedef int16_t celt_int16;
   typedef u_int16_t celt_uint16;`

**Comment by Krzmbrzl** [comment]
Yeah the typedefs and the logic around them is broken. However, it seems like a simple update of the submodule should be enough to fix this issue.
I just didn't get around to fixing this yet
